File: backend/education/src/main/java/com/braintrust/education/application/dtos/dtos/EnrollmentDTO.java
package com.braintrust.education.application.dtos.dtos;

public record EnrollmentDTO(
        String id,
        String courseId,
        String courseName,
        String studentId,
        String studentName,
        String studentEmail,
        String studentRefId,
        String enrollmentDate,
        String status,
        GradeDTO finalGrade
) {}

File: frontend/app/shared/dtos/enrollment.dto.ts
export interface EnrollmentDTO {
    id: string;
    courseId: string;
    courseName: string;
    studentId: string;
    studentName: string;
    studentEmail: string;
    studentRefId: string;
    enrollmentDate: string;
    status: string;
    finalGrade: GradeDTO | null;
}

export interface GradeDTO {
    grade: number;
    feedback: string;
    gradedBy: string;
    gradedDate: string;
}

export interface FinalGradeDTO {
    value: string;
    maxScore: string;
    percentage: string;
}